components/esp_modem/include/cxx_include/esp_modem_cmux.hpp

```diff
@@ -64,6 +64,13 @@ class CMux {
      */
     [[nodiscard]] bool init();
 
+    /**
+     * @brief   Closes CMux protocol and ejects attached terminal and buffer
+     * @return  nullptr on failure
+     *          tuple of the original terminal and buffer on success
+     */
+    std::tuple<std::unique_ptr<Terminal>, std::unique_ptr<uint8_t[]>, size_t> deinit_and_eject();
```

Comment:
I'm afraid we cannot make it an integral part of the `Terminal` (since it's shared with the CMUX-like terminal).

What we could do though is to bundle the `unique-ptr` and the size into a type, I'm on it.

Reply:
Created `unique_buffer` to hold the pointer, size and offset, and is `std::move`-able so we can simply transfer ownership between `CMux` and `DTE`